app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt

```diff
@@ -20,16 +32,33 @@ import org.fossasia.openevent.general.utils.Utils.hideSoftKeyboard
 import org.fossasia.openevent.general.utils.Utils.show
 import org.fossasia.openevent.general.utils.Utils.progressDialog
 import org.fossasia.openevent.general.utils.extensions.nonNull
-import org.jetbrains.anko.design.longSnackbar
-import org.jetbrains.anko.design.snackbar
 import org.koin.androidx.viewmodel.ext.android.sharedViewModel
 import org.koin.androidx.viewmodel.ext.android.viewModel
+import timber.log.Timber
+import com.facebook.GraphRequest
+import kotlinx.android.synthetic.main.dialog_password.view.textInputLayoutPassword
+import kotlinx.android.synthetic.main.dialog_password.view.textInputLayoutConfirmPassword
+import kotlinx.android.synthetic.main.dialog_password.view.password
+import kotlinx.android.synthetic.main.dialog_password.view.confirmPassword
+import org.fossasia.openevent.general.event.EVENT_DETAIL_FRAGMENT
+import org.fossasia.openevent.general.notification.NOTIFICATION_FRAGMENT
+import org.fossasia.openevent.general.order.ORDERS_FRAGMENT
+import org.fossasia.openevent.general.ticket.TICKETS_FRAGMNET
+import org.jetbrains.anko.design.longSnackbar
+import org.jetbrains.anko.design.snackbar
+
+private const val EMAIL = "email"
+private const val FIRST_NAME = "first_name"
+private const val LAST_NAME = "last_name"
 
 class AuthFragment : Fragment() {
     private lateinit var rootView: View
     private val authViewModel by viewModel<AuthViewModel>()
     private val safeArgs: AuthFragmentArgs by navArgs()
     private val smartAuthViewModel by sharedViewModel<SmartAuthViewModel>()
+    private val callbackManager = CallbackManager.Factory.create()
+    private lateinit var signUp: SignUp
+    private var fbLogIn = false
```

Comment:
I suggest keeping parameter about the logic inside viewModel

Reply:
Here we need to register `callbackManager` with the fragment and handle activity results so it will complex to use view model. I will try it. :+1: